x-pack/plugins/lens/public/editor_frame_service/editor_frame/suggestion_helpers.ts

```diff
@@ -211,26 +211,30 @@ function getVisualizationSuggestions(
   isFromContext?: boolean,
   activeData?: Record<string, Datatable>
 ) {
-  return visualization
-    .getSuggestions({
-      table,
-      state: currentVisualizationState,
-      keptLayerIds: datasourceSuggestion.keptLayerIds,
-      subVisualizationId,
-      mainPalette,
-      isFromContext,
-      activeData,
-    })
-    .map(({ state, ...visualizationSuggestion }) => ({
-      ...visualizationSuggestion,
-      visualizationId,
-      visualizationState: state,
-      keptLayerIds: datasourceSuggestion.keptLayerIds,
-      datasourceState: datasourceSuggestion.state,
-      datasourceId: datasourceSuggestion.datasourceId,
-      columns: table.columns.length,
-      changeType: table.changeType,
-    }));
+  try {
+    return visualization
+      .getSuggestions({
+        table,
+        state: currentVisualizationState,
+        keptLayerIds: datasourceSuggestion.keptLayerIds,
+        subVisualizationId,
+        mainPalette,
+        isFromContext,
+        activeData,
+      })
+      .map(({ state, ...visualizationSuggestion }) => ({
+        ...visualizationSuggestion,
+        visualizationId,
+        visualizationState: state,
+        keptLayerIds: datasourceSuggestion.keptLayerIds,
+        datasourceState: datasourceSuggestion.state,
+        datasourceId: datasourceSuggestion.datasourceId,
+        columns: table.columns.length,
+        changeType: table.changeType,
+      }));
+  } catch (e) {
+    return [];
```

Comment:
Does it make sense to fail so quietly here? We don't want the editor to break when an error is thrown. On the other hand, we don't want bugs to be so hidden that we don't catch them before release.